Algorithm.Framework/Alphas/RsiAlphaModel.cs

```diff
@@ -0,0 +1,275 @@
+        /// Updates this alpha model with the latest data from the algorithm.
+        /// This is called each time the algorithm receives data for subscribed securities
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <param name="data">The new data available</param>
+        /// <returns>The new insights generated</returns>
+        public IEnumerable<Insight> Update(QCAlgorithmFramework algorithm, Slice data)
+        {
+            foreach (var kvp in _symbolDataBySymbol)
+            {
+                var symbol = kvp.Key;
+                var rsi = kvp.Value.RSI;
+                var previousState = kvp.Value.State;
+                var state = GetState(rsi, previousState);
+
+                if (state != previousState && rsi.IsReady)
+                {
+                    switch (state)
+                    {
+                        case State.TrippedLow:
+                            yield return new Insight(symbol, InsightType.Price, InsightDirection.Up, _parameters.PredictionInterval);
+                            break;
+
+                        case State.TrippedHigh:
+                            yield return new Insight(symbol, InsightType.Price, InsightDirection.Down, _parameters.PredictionInterval);
+                            break;
+                    }
+                }
+
+                kvp.Value.State = state;
+            }
+        }
+
+        /// <summary>
+        /// Cleans out old security data and initializes the RSI for any newly added securities.
+        /// This functional also seeds any new indicators using a history request.
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance that experienced the change in securities</param>
+        /// <param name="changes">The security additions and removals from the algorithm</param>
+        public void OnSecuritiesChanged(QCAlgorithmFramework algorithm, SecurityChanges changes)
+        {
+            // clean up data for removed securities
+            if (changes.RemovedSecurities.Count > 0)
+            {
+                var removed = changes.RemovedSecurities.ToHashSet(x => x.Symbol);
+                foreach (var subscription in algorithm.SubscriptionManager.Subscriptions)
+                {
+                    if (removed.Contains(subscription.Symbol))
+                    {
+                        _symbolDataBySymbol.Remove(subscription.Symbol);
+                        subscription.Consolidators.Clear();
+                    }
+                }
+            }
+
+            // initialize data for added securities
+            if (changes.AddedSecurities.Count > 0)
+            {
+                foreach (var added in changes.AddedSecurities)
+                {
+                    if (!_symbolDataBySymbol.ContainsKey(added.Symbol))
+                    {
+                        var rsi = algorithm.RSI(added.Symbol, 14, MovingAverageType.Wilders, Resolution.Daily);
```

Comment:
Should this use the _parameters.Period / .Resolution?

Reply:
Absolutely. This has been fixed.